### ISSUE TITLE
View composers not firing for included files

### ISSUE THREAD

**Issue by JayBizzle** [ytake/Laravel.Smarty#37]
I have this simple test file...

``` html
<!-- base.tpl -->

{include file='main-nav.tpl'}
```

Then i have this view composer...

``` php
<?php

namespace App\Providers;

use App\Repositories\Eloquent\UsersRepository;
use Illuminate\Support\ServiceProvider;

class ViewComposerServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap the application services.
     *
     * @return void
     */
    public function boot(UsersRepository $users)
    {
        $this->app['view']->composer('main-nav', function ($view) use ($users) {
            $view->with('userCount', $users->count());
        });
    }

    /**
     * Register the application services.
     *
     * @return void
     */
    public function register()
    {
        //
    }
}
```

...but it seems that view composer aren't fired for any files that are included like this.

Looking at the output of in Debugbar, the included `main-nav.tpl` doesn't get logged as rendered.

![screen shot 2016-06-08 at 21 21 04](https://cloud.githubusercontent.com/assets/340752/15909648/2411efba-2dbf-11e6-90d1-b4e2ffdab933.png)

**Comment by unstoppablecarl** [comment]
View composers are not really supported. You can do 
```php
<?php
$this->app['view']->composer('*', ...
```

**Comment by JayBizzle** [comment]
@unstoppablecarl Yes, that is one method that does work, but is not ideal. We are using some packages that still, internally, use blade templates i.e. https://github.com/davejamesmiller/laravel-breadcrumbs

These internal templates trigger the view composers, so if your view composer runs a query, like ours does, you can end up running the same query multiple time needlessly.

**Comment by unstoppablecarl** [comment]
@JayBizzle I'm not sure if there is a reasonable way to bridge view composers between blade and smarty. 

Caching the viewcomposer values may be a reasonable solution.

**Comment by JayBizzle** [comment]
I thought as much. I had a dig around the source code, but couldn't see any easy way to hook into the view composers.

**Comment by JayBizzle** [comment]
Wonder if this sheds any light on achieving this? Only had a quick glance

https://github.com/rcrowe/TwigBridge/blob/2ff40018d7b4e996eed7d8e771cc052de909c7da/src/Twig/Template.php#L51-L71

**Comment by JayBizzle** [comment]
@ytake was this branch https://github.com/ytake/Laravel.Smarty/tree/feature-smarty-include-events the start of an attempt at resolving this issue?

**Comment by ytake** [comment]
I’m trying to solve it.

**Comment by ytake** [comment]
released v2.1.11

**Comment by JayBizzle** [comment]
Excellent. Won't be able to test this out for a couple of days, but will report back when I have :+1:

**Comment by JayBizzle** [comment]
Fix in [v2.1.12](https://github.com/ytake/Laravel.Smarty/releases/tag/2.1.12)
